modules/core/src/main/java/org/apache/ignite/internal/util/typedef/X.java

```diff
@@ -464,37 +469,22 @@ private static void cloneField(Map<Integer, Integer> identityIdxs, List<Object>
      * @return {@code True} if one of the causing exception is an instance of passed in classes,
      *      {@code false} otherwise.
      */
-    @SafeVarargs
-    public static boolean hasCause(@Nullable Throwable t, @Nullable String msg, @Nullable Class<?>... cls) {
+    public static boolean hasCause(
+        @Nullable Throwable t,
+        @Nullable String msg,
+        @Nullable Class<?>... cls
+    ) {
         if (t == null || F.isEmpty(cls))
             return false;
 
-        assert cls != null;
+        Predicate<Throwable> pred = thr -> Arrays.stream(cls)
```

Comment:
This method can be used in hot methods, it should be converted to methods with loops.

Reply:
Use indexed loops whenever possible.

Reply:
changed to raw loop. Thank you.